include/svnlook.php

```diff
@@ -1039,11 +1039,11 @@ function getList($path, $rev = 0, $peg = '') {
 				$rev = $headlog->entries[0]->rev;
 		}
 
-		$cmd = $this->svnCommandString('list --xml', $path, $rev, $peg);
+		$cmd = $this->svnCommandString('list -R --xml', $path, $rev, $peg);
 		$this->_xmlParseCmdOutput($cmd, 'listStartElement', 'listEndElement', 'listCharacterData');
 
 		// Sort the entries into alphabetical order
-		usort($curList->entries, '_listSort');
+		/*usort($curList->entries, '_listSort');*/
```

Comment:
Why did you remove the sorting? Many user interfaces sort directory and file entries according some criteria instead of relying on arbitrary order of the underlying file system or whatever.

Reply:
When i take the xml output out it has the correct order of directories and the files. First the directory then the files under it and then so on. 
When i do this sort it puts all the directories at top and then fhe files in an unrecognizable order. Where the xml has the proper order printed out. So i removed the sorting.

Reply:
I did some investigation. This function will not work in this case as it is fed the whole list of directories and files by this method. 
Previously it was being fed level by level. First the root level, then the folder that is clicked level and so on... So it was never organizing everything, only each level. 
Alphabetically sorting the entire list while still keeping level information will be tough as the xml output doesnt provide any level information. We calculate level by slicing the folders for each file that is fed.

Reply:
So whats your conclusion, remove the entire thing as I suggested? I'm fine with applying alphabetical sort always by default, without any further config. You only need to confirm that things are sorted alphabetically in your case with loading everything recursively AND with the former behaviour loading one level only.

Reply:
I am not sure in the full recursive scenario i will be able to arrange alphabetically. Let me try it first i will come back here after that. Then we can come to conclusion.

Reply:
We might be talking about different things: I DO NOT recommend alphabetical sorting ACROSS dirs and files. It should be dirs first, than files and within both alphabetically always. That's how I understood your following sentence:

> First the directory then the files under it and then so on.

But the XML doesn't care about alphabetical sort within each group?

Reply:
Yes the XML doesn't care about sorting. There is no group in xml. Sorting there does sorting across dirs and files. There is no way to avoid that. 
Unless we do further processing on it. There is no way for recursive xml to first sort the dirs and then the files like it is being currently done. 

Here is a snapshot of xml. 
![image](https://user-images.githubusercontent.com/4973182/92223949-d308fd00-eebe-11ea-8b5f-9cd4714442d0.png)

Reply:
@ams-tschoening Now i understood what you meant. I have made the sorting by alphabetical order by default for when the repository load all at once.